front/src/applications/operationalStudies/components/SimulationResults/SpeedSpaceChart/SpeedSpaceChart.jsx

```diff
@@ -189,6 +195,31 @@ export default function SpeedSpaceChart(props) {
     };
   }, [chart, trainSimulation, localSettings, resetChart, rotate]);
 
+  // to be removed after creating power restriction module
+  useEffect(() => {
+    setRestrictionPower('');
+    trainSimulation.electrificationConditions.forEach((elem) => {
+      if (elem.used_restriction)
```

Comment:
I don't see used_restriction in any type in the code ? We should try to type everything

Reply:
done